static/app/components/clippedBox.tsx

```diff
@@ -29,7 +56,18 @@ function revealAndDisconnectObserver({
     return;
   }
 
-  wrapperRef.current.style.maxHeight = '9999px';
+  const clearMaxHeight = () => {
+    if (wrapperRef.current) {
+      wrapperRef.current.style.maxHeight = 'none';
+    }
+  };
+
+  wrapperRef.current.addEventListener('transitionend', clearMaxHeight, {once: true});
+
+  const revealedWrapperHeight =
+    (contentRef.current?.clientHeight || 9999) + calculateAddedHeight({wrapperRef});
+
+  wrapperRef.current.style.maxHeight = `${revealedWrapperHeight}px`;
```

Comment:
I wonder if we can be smarter about this, querying the DOM on L68 will cause layout thrashing which can snowball into a perf issue if a page has many of these, but I think we should worry about this later since we have profiling instrumentation and can monitor this.